src/models/selection.ts

```diff
@@ -1,15 +1,23 @@
 import { RunGraphNode, Artifact } from '@/models'
 
 export type NodeSelection = {
-  id: string,
   kind: RunGraphNode['kind'],
+  id: string,
 }
 
 export type ArtifactSelection = {
   kind: 'artifact',
   id: string,
 }
 
-export type GraphItemSelection = NodeSelection | ArtifactSelection
+export type ArtifactClusterSelection = {
+  kind: 'artifactCluster',
+  ids: string[],
+}
```

Comment:
super nit but this might be a nice pattern since we'll have other things that are clustered
```suggestion
export type ArtifactsSelection = {
  kind: 'artifacts',
  ids: string[],
}
```